grpc.go

```diff
@@ -848,43 +848,48 @@ func (a *agentGRPC) UpdateContainer(ctx context.Context, req *pb.UpdateContainer
 		return emptyResp, err
 	}
 
-	config := c.container.Config()
```

Comment:
`c.container.Config()` returns a copy of type `configs.Config`. Why do you need to duplicate this copy ?
I would assume that if something goes wrong during `c.container.Set(config)`, this is libcontainer's responsibility to handle a proper rollback.
The issue is not very clear to me, could you elaborate ?

Reply:
`c.container.Config()` returns a copy of the non-pointer elements in `configs.Config`, unfortunately [Cgroup](https://github.com/opencontainers/runc/blob/master/libcontainer/configs/config.go#L128) and [Resources](https://github.com/opencontainers/runc/blob/master/libcontainer/configs/cgroup_linux.go#L31) are pointers, if one of them is modified (like currently we do) , if `c.container.Set` fails it won't be able to rollback because of the container cgroup was modified, see https://github.com/opencontainers/runc/blob/master/libcontainer/container_linux.go#L205

I'll add more comments in the code